File: DPLABFINAL/DPLabFinal/src/IssueRepository.java
import java.util.HashMap;
import java.util.Map;

public class IssueRepository {
    private Map<String,Issue>issues;

    public IssueRepository(){
        this.issues=new HashMap<>();
    }

    public void addIssue(Issue issue){
        issues.put(issue.getId(),issue);
    }

    public Issue getIssue(String id){
        return issues.get(id);
    }

    public Issue removeIssue(String id)
    {
        return issues.remove(id);
    }

    public Map<String,Issue> getAllIssues(){
        return issues;
    }


}

File: TaskManager/TaskManager/Repositories/IssueRepository.cs
using System;
using System.Collections.Generic;
using TaskManager.Models;

namespace TaskManager.Repositories
{
    public class IssueRepository
    {
        private readonly Dictionary<int, Issue> _issues = new Dictionary<int, Issue>();
        private int _nextId = 1;

        public void AddIssue(Issue issue)
        {
            issue.Id = _nextId++;
            issue.CreationDate = DateTime.Now;
            issue.LastModifiedDate = DateTime.Now;
            _issues[issue.Id] = issue;}

        public Issue GetIssue(int id)
        {
            return _issues.ContainsKey(id) ? _issues[id] : null;
        }

        public void UpdateIssue(Issue issue)
        {
            issue.LastModifiedDate = DateTime.Now;
            _issues[issue.Id] = issue;
        }

        public IEnumerable<Issue> GetAllIssues()
        {
            return _issues.Values;  
        }
        }
}